# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1295720983 18000
#      Sat Jan 22 13:29:43 2011 -0500
# Node ID dc11166fcddc330d98e2a4a1bed8fe7ed333bb6d
# Parent  da9afc5c4cfbe4d18d1e3ea941edcb91b7db42d5
subplot.m: Set "box" to "on" by default (same as Matlab).

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,11 @@
 2011-01-22  Ben Abbott  <bpabbott@mac.com>
 
+	* plot/subplot.m: Set "box" to "on" by default (same as Matlab).
 	* plot/cylinder.m: Remove console output from demo.
 
 2011-01-22  Konstantinos Poulios  <logari81@googlemail.com>
 
 	* plot/plotyy.m: Set box property to off to allow both y-axes colors
 	to be visible for OpenGL backends.
 
 2011-01-22  John W. Eaton  <jwe@octave.org>
diff --git a/scripts/plot/subplot.m b/scripts/plot/subplot.m
--- a/scripts/plot/subplot.m
+++ b/scripts/plot/subplot.m
@@ -160,17 +160,18 @@ function h = subplot (rows, cols, index,
     endfor
 
     if (found)
       set (cf, "currentaxes", tmp);
     else
       pos = subplot_position (rows, cols, index, "outerposition", units);
       pos2 = subplot_position (rows, cols, index, "position", units);
       tmp = axes ("outerposition", pos, "position", pos2,
-                  "activepositionproperty", "outerposition");
+                  "activepositionproperty", "outerposition",
+                  "box", "off");
     endif
 
     if (align_axes && strcmp (get (cf, "__graphics_toolkit__"), "gnuplot"))
       set (tmp, "activepositionproperty", "position");
     endif
 
   unwind_protect_cleanup
     set (0, "defaultaxesunits", units);
